Offset analysis updates

tab-cmd opened the pull request:
# Overview

Add support for determining offsets between timing verification Tasks (Ex. RSVPTimingVerificationCalibration) and RawData with a photodiode trigger column. This is useful for setting up new systems and preventing errors before an experiment begins.

## Ticket

- https://www.pivotaltracker.com/story/show/178696232
- https://www.pivotaltracker.com/story/show/187636698

## Contributions

- Fixed bug with Matrix Timing Verification
- Moved offset script to a module in helpers. 

## Test

- `make test-all`
- `make offset`
- `make offset-recommend`

## Documentation

- Are documentation updates required? In-line, README, or [documentation](https://github.com/BciPy/bcipy.github.io)? In-line and README updates

## Changelog

- Is the CHANGELOG.md updated with your detailed changes? Yes

tab-cmd committed:
move offset to a module (IP)

tab-cmd committed:
Move offset to a module and add tests / typing

add to README and CHANGELOG

Carsonthemonkey commented on bcipy/helpers/offset.py:
```diff
@@ -0,0 +1,336 @@
+    if recommend_static:
+        # test for normality
+        _, p_value = normaltest(diffs)
+        print(f'{describe(diffs)}')
```
Is the f-string necessary here?

Carsonthemonkey reviewed (approved):
Looks good to me!

lawhead reviewed (approved):
This is a nice contribution and much needed. Thanks!

tab-cmd committed:
Update offset to include correction for false diode responses

Add new tests, use the stim number to determine how many stimuli before skipping.

tab-cmd committed:
lint
